### ISSUE TITLE
Questions about tsc / rollup configuration.  How to watch / rebuild for npm link scenario

### ISSUE THREAD

**Issue by mattmazzola** [alexjoverm/typescript-library-starter#176]
I came across scenario that doesn't seem supported out of box with typescript-library-starter and wanted to better understand the decisions for current configuration and if there is alternate method to achieve this

## Scenario:
Developer has linked the library built with typescript-library-starter into other consuming project and is actively making changes.  Ideally the developer can update the source of the linked package and see the changes immediately in consuming package; however, the package.json points to the `/dist` folder and from what I see there isn't a way to easily rebuild so the developer must manually run `npm run build` This is not ideal because it slows down the feedback loop as the developer manually runs the command and also the `npm run build` does other things like building docs that isn't needed.

In normal projects we can run tsc with watch mode and it will recompile on save enabling a nice workflow across packages. However, typescript library starter re-processes the output of tsc with rollup:

`"build": "tsc --module commonjs --outDir dist/lib && rollup -c rollup.config.ts ...`
so it's not as simple as adding another command such as:
"watch": "tsc --watch"
as this will only put the files into the `dist/es` folder instead of `dist` which the consuming library needs.

All the above brings up the questions about tsc configuration.
By default the tsconfig.json has:
```
"target": "es5",
"module":"es2015",
...
"outDir": "dist/es",
```

However, when running the actual build it uses inline parameters to override the tsconfig:
`tsc --module commonjs --outDir dist/lib`

And then this is recompiled by rollup for what I assume to be tree-shaking.

## Questions:
**1. What is the use case for putting the default to dist/es as es2015 format?**
Is this output used by anything? If not, why not move the inline tsc parameters into tsconfig.json?

**2. Is it correct that rollup is re-writing the files outputted to `dist/lib`?**
From looking at rollup.config.ts it seems that it's entry point is still `src/${libraryname}.ts` so I don't see why `dist/lib` is needed, but given that tsc --outDirt dist/lib was intentionally called before rollup I assume it is required.

There is a comment that says:
> // Allow bundling cjs modules (unlike webpack, rollup doesn't understand cjs)

but did is tsc configured to output into commonjs if rollup can't understand it?

**3. Rollup says it has a watch on `src/**`** 
It seems I could add an npm script to achieve goals:
`"watch": "rollup -c rollup.config.ts --watch"`
which would by default watch src and recompile to /dist

**4. What is alternative/better approach for re-building on save for npm linking?**

Overall I would say having some discussion like these can help users learn / understand reasoning behind the current state which will help them feel comfortable making changes where needed.

**Comment by alexjoverm** [comment]
> 1. What is the use case for putting the default to dist/es as es2015 format?
Answered in #134. Although, see next point.

> 2. Is it correct that rollup is re-writing the files outputted to dist/lib?
This seems to be an issue, this should be cjs but it is es right now. Looking into it.


> 3. Rollup says it has a watch on src/**
> 4. What is alternative/better approach for re-building on save for npm linking?
This should be working already by running `npm start`, check the start script :)

**Comment by alexjoverm** [comment]
@mattmazzola I've done and merged some PR #186 #183 #185 #184 that I think they solve all these issues. 

Since multi-module libraries (`import something from 'mylib/dist/lib/something'`) don't makes sense for tree-shaking, I've preferred in #186 to output them in cjs so that they're usable also in nodejs.

Please feel free to open a new issue if you find anything wrong.
